### PR TITLE
Adding instance pools swagger

### PR DESCRIPTION
**Pull request by johnpaulkee**
### Instance pools
Adding a new instancePools.json swagger file and its examples.
Instance pools are a new deployment option for managed instances, enabling pre-provisioning of compute as a customer step. This enables customers to target managed instances in an instance pool and have predictable provisioning times.
This instancePools.json swagger file is hooked up with the basic CRUD operations for an instance pool.
- Create an instance pool
- List / Get an instance pool
- Delete an instance pool

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by veronicagg** [comment]
@kpajdzik assigning this one to you for review, thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: rerquired -> required

Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: instacne -> instance

**Review by nschonni — COMMENTED** [review]
Not sure if the typos are actually in the exception response

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: acked, found out that they are in the exception response in general. I'll fix it.

Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: acked.

Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Fixed typo

Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Fixed typo

**Comment by johnpaulkee** [comment]
Fixed typo in examples @nschonni .
Adding @jaredmoo to help with the review.

**Comment by johnpaulkee** [comment]
Note I don't plan on making updates to the README.md file just yet,
I intend on merging the instancePools json and its examples alone so we can deploy ARM Manifest and test instance pool CRUD in production using .NET SDK.

**Comment by kpajdzik** [comment]
If you haven't taken a look at our new onboarding experience at [OpenAPIHub](http://aka.ms/openapihub), it's a convenient way to create your PR when you're copying from an existing API version or when you're editing your existing specs. If you have any feedback or questions, feel free to use the feedback button on top of the site for help. Thanks!

**Comment by johnpaulkee** [comment]
Taking a look at OpenAPIHub @kpajdzik, thanks 👍 
Can you help me re-run the linter diff, not sure why it's breaking after I undid changes to README.md

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (kpajdzik) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Can this be expanded? It's not helpful in current form.

Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Discussed we always return empty list if no instance pools found in resource group.
We do similar in https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json#L45

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Please add a subscription-level list API, i.e. `/subscriptions/{subscriptionId}/providers/Microsoft.Sql/instancePools`

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: I assume you will be adding a linked access check for this property?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (KrisBash) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Can this be an enum? Is the list of allowed values finite?

**Review by KrisBash — CHANGES_REQUESTED** [review]
A few minor changes requested. thanks

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: The list of values are finite for now [LicenseIncluded / BasePrice]
We are following managedInstances design.

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json#L341

Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Enum could be nice, but making this consistent between managedInstances and instancePools would be desired.
Let me know if you think making it inconsistent is okay.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Yup, we will be adding a linked access check for subnet in ARM Manifest.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: @tomapaunovic is making changes in this area for MI. Please sync with him. They really should both be using enum (model as string = true).

**Review by jaredmoo — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Oh awesome, thanks for letting us know @jaredmoo.
I'll sync with @tomapaunovic about this. Can probably re-use his license types once he has checked in.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Will add.

**Review by kpajdzik — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Made updates in latest commit.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json
Comment (johnpaulkee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/instancePools.json]: Added. Note I'm adding DO NOT MERGE back to title. I'm waiting for approval for check in for API changes but didn't want to block swagger review. Once its checked in I'll get rid of DO NOT MERGE again.

**Comment by johnpaulkee** [comment]
Thanks for the reviews everyone.

**Review by KrisBash — APPROVED** [review]
Thanks

**Comment by kpajdzik** [comment]
@johnpaulkee, seems like it's signed off from ARM and our side as well. When do you want to merge it?

Also, if you're not aware, GitHub introduce Pull Request Drafts which are better solution than "[Do not merge]" tags in the title (they can't be merged by accident). More information: https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by johnpaulkee** [comment]
Hey @kpajdzik, ready to merge :)
We probably won't add changes to README.md until a separate PR with changes to managedInstances.

Also, that's a cool new PR draft feature github added. Will use it next time.
Thanks! 👍

**Comment by kpajdzik** [comment]
@johnpaulkee , Do you want me to merge it?

**Comment by kpajdzik** [comment]
@johnpaulkee, If you need to release the new SDK, please create a request on the portal mentioned above (http://aka.ms/openapihub).

**Pull request merged**
